sklearn/ensemble/bagging.py

```diff
@@ -33,21 +33,55 @@
 MAX_INT = np.iinfo(np.int32).max
 
 
+def _generate_mask_from_indices(indices, mask_length, inverse=False):
+    """Private function used to convert list of indices to boolean mask"""
+    if mask_length <= np.max(indices):
+        raise ValueError("mask_length must be greater than max(indices)")
+
+    if inverse:
+        mask = np.ones(mask_length, dtype=np.bool)
+        mask[indices] = False
+    else:
+        mask = np.zeros(mask_length, dtype=np.bool)
+        mask[indices] = True
+
+    return mask
+
+
+def _generate_indices(random_state, bootstrap, n_population, n_samples):
+    """Private function used to draw randomly sampled indices."""
+    # Draw sample indices
+    if bootstrap:
+        indices = random_state.randint(0, n_population, n_samples)
+    else:
+        indices = sample_without_replacement(n_population, n_samples, 
+                                             random_state=random_state)
+
+    return indices
+
+
+def _generate_bagging_indices(random_state, bootstrap_features, bootstrap_samples, 
+                              n_features, n_samples, max_features, max_samples):
+    """Private function used to randomly draw feature and sample indices."""
```

Comment:
I wrote it this way to be consistent with `_parallel_build_estimators`, `_parallel_predict_proba`, etc. 

Should I edit the docstrings for those functions too?